src/cryptonote_core/blockchain.cpp

```diff
@@ -4698,12 +4698,21 @@ bool Blockchain::calc_batched_governance_reward(uint64_t height, uint64_t &rewar
   // Ignore governance reward and payout instead the last
   // GOVERNANCE_BLOCK_REWARD_INTERVAL number of blocks governance rewards.  We
   // come back for this height's rewards in the next interval. The reward is
-  // 0 if it's not time to pay out the batched payments
+  // 0 if it's not time to pay out the batched payments (in which case we
+  // already returned, above).
 
   const cryptonote::config_t &network = cryptonote::get_config(nettype(), hard_fork_version);
   size_t num_blocks                   = network.GOVERNANCE_REWARD_INTERVAL_IN_BLOCKS;
-  uint64_t start_height               = height - num_blocks;
 
+  // Fixed reward starting at HF15
+  if (hard_fork_version >= network_version_15_lns)
```

Comment:
There's an edge case here on the boundary straddling HF14, H15 and HF15, HF16. Supposing the payout interval was 100 blocks, and HF14 ran from height 0->49, then HF15 starts at 50->100, you would want to pay 50 blocks of HF14 and 50 blocks of HF15.

Reply:
I deliberately don't want to do that because it's a nuisance and doesn't really make a significant difference.  If someone objects strongly that it overpays for a few days around the fork then LF can deal with it (e.g. by burning the difference).